File: tests/fixtures/dotnet_solution/src/Infrastructure/Repositories/UserRepository.cs
using Microsoft.EntityFrameworkCore;
using Acme.Domain;

namespace Acme.Infrastructure;

public class UserRepository : IUserRepository
{
    private readonly AppDbContext _db;
    public UserRepository(AppDbContext db) { _db = db; }

    public Task<User?> FindAsync(string email) =>
        _db.Users.FirstOrDefaultAsync(u => u.Email == email);

    public Task AddAsync(User user)
    {
        _db.Users.Add(user);
        return _db.SaveChangesAsync();
    }
}

File: tests/fixtures/vbnet_solution/src/Infrastructure/UserRepository.vb
Imports Microsoft.EntityFrameworkCore
Imports Acme.Domain

Public Class UserRepository
    Implements IUserRepository

    Private ReadOnly _db As AppDbContext

    Public Sub New(db As AppDbContext)
        _db = db
    End Sub

    Public Function FindAsync(email As String) As Task(Of User) Implements IUserRepository.FindAsync
        Return _db.Users.FirstOrDefaultAsync(Function(u) u.Email = email)
    End Function

    Public Function AddAsync(user As User) As Task Implements IUserRepository.AddAsync
        _db.Users.Add(user)
        Return _db.SaveChangesAsync()
    End Function
End Class
